File: src/main/java/com/sfh/pokeRogueBot/model/modifier/MoveToModifierResult.java
package com.sfh.pokeRogueBot.model.modifier;

import com.sfh.pokeRogueBot.model.enums.ModifierApplyType;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public class MoveToModifierResult {

    private int rowIndex;
    private int columnIndex;
    private int pokemonIndexToSwitchTo;
    private String itemName;
    private ModifierApplyType modifierApplyType;

    @Override
    public String toString() {
        return itemName + " for pokemon at index " + pokemonIndexToSwitchTo;
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/MoveToModifierResult.kt
package com.sfh.pokeRogueBot.model.modifier

data class MoveToModifierResult(
    val rowIndex: Int,
    val columnIndex: Int,
    val pokemonIndexToSwitchTo: Int,
    val itemName: String
) {
    override fun toString(): String {
        return "$itemName for pokemon at index $pokemonIndexToSwitchTo"
    }
}